src/webportal/src/app/job-submission/components/data/data-component.jsx

```diff
@@ -46,8 +46,14 @@ function reducer(state, action) {
 }
 
 export const DataComponent = React.memo((props) => {
-  // add pylonAddress to .env for local debug
-  const hdfsHost = config.hasOwnProperty('pylonAddress') ? getHostNameFromUrl(config.pylonAddress) : window.location.hostname;
+  const envsubRegex = /^\${.*}$/;
+  let hdfsHost;
+  if (!config.pylonAddress || envsubRegex.test(config.pylonAddress)) {
+    hdfsHost = window.location.hostname;
+  } else {
+    // add pylonAddress to .env for local debug
+    hdfsHost = getHostNameFromUrl(config.pylonAddress);
+  }
   // TODO: add a judgement whether pylon is ready
   const hdfsClient = new WebHDFSClient(hdfsHost);
```

Comment:
Should put port and path to the parameters?
If we use customized hdfs url. The port may different with the default value.

Reply:
yes, I will put the parameters outside